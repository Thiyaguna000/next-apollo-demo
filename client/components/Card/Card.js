import React from "react";
import styles from "./card.module.css";
import { useState } from "react";

const Card = ({ data }) => {
    return(
    <>
        <div className={styles.dataContainer}>
            <h2>{data.name}</h2>
            <h2>{data.email}</h2>
            <h2>{data.address}</h2>
        </div>
    </>
        
    )
};

export default Card;