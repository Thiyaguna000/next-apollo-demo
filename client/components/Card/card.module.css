.dataContainer {
    padding: 10px;
    border: 1px solid #C7C8D9;
    margin: 6px;
    font-family: 'Open Sans', sans-serif;
}

.header{
    text-align: center;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
}

.subHeader{
    color:#555770;
}

.location{
    color: #000000;
    font-weight: 500;
}