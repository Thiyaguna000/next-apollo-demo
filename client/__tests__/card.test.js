import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Card from '../components/Card/Card';

afterEach(cleanup);

const dummyData = [
    {
        name: 'name',
        location: 'location',
        webPages: 'webPages'
    }
]

it('renders card', () => {
    const tree = renderer
      .create(<Card data={dummyData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Checking for Label renders in Card Component', () => {
    const {getByText} = render(
      <Card data={dummyData} />,
    );
  
    expect(getByText("Website:")).toBeTruthy();
    expect(getByText("Location:")).toBeTruthy();
  });

