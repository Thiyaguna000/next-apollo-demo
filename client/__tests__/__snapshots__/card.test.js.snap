// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card 1`] = `
<div
  className="cardData"
>
  <h2
    className="header"
  />
  <p
    className="subHeader"
  >
    Location:
    <span
      className="location"
    />
  </p>
  <p>
    Website: 
    <a
      target="_blank"
    />
  </p>
</div>
`;
